static_frame/core/util.py

```diff
@@ -415,61 +412,60 @@ def depth_level_from_specifier(
         tp.Dict[TLabel, TDtypeSpecifier]
         ]]
 
-# specifiers that are equivalent to object
-DTYPE_SPECIFIERS_OBJECT = {DTYPE_OBJECT, object, tuple}
-
 TDepthLevelSpecifier = tp.Union[int, tp.List[int], slice, np.ndarray, None]
 TDepthLevel = tp.Union[int, tp.List[int]]
 
-CallableToIterType = tp.Callable[[], tp.Iterable[tp.Any]]
+TCallableToIter = tp.Callable[[], tp.Iterable[tp.Any]]
 
-IndexSpecifier = tp.Union[int, TLabel] # specify a postiion in an index
-IndexInitializer = tp.Union[
+TIndexSpecifier = tp.Union[int, TLabel] # specify a position in an index
+TIndexInitializer = tp.Union[
         'IndexBase',
         tp.Iterable[TLabel],
         tp.Iterable[tp.Sequence[TLabel]], # only for IndexHierarchy
         # tp.Type['IndexAutoFactory'],
         ]
 
-IndexConstructor = tp.Optional[tp.Union[tp.Callable[..., 'IndexBase'], tp.Callable[..., 'Index'], tp.Type['Index']]]
-IndexConstructors = tp.Union[IndexConstructor,
-        tp.Sequence[IndexConstructor],
-        tp.Iterator[IndexConstructor],
+TIndexCtor = tp.Union[tp.Callable[..., 'IndexBase'], tp.Callable[..., 'Index'], tp.Type['Index']]
```

Comment:
Why not the additional specifications for `IndexHierarchy`?

Reply:
I think it is already covered by `tp.Callable[..., 'IndexBase']`...